src/script/lua_api/l_util.cpp

```diff
@@ -245,6 +246,34 @@ int ModApiUtil::l_get_hit_params(lua_State *L)
 	return 1;
 }
 
+// check_password_entry(name, entry, password)
+int ModApiUtil::l_check_password_entry(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+	std::string name = luaL_checkstring(L, 1);
+	std::string entry = luaL_checkstring(L, 2);
+	std::string password = luaL_checkstring(L, 3);
+
+	if (base64_is_valid(entry)) {
+		std::string hash = translate_password(name, password);
+		lua_pushboolean(L, hash == entry);
+		return 1;
+	}
+
+	std::string salt;
+	std::string verifier;
+
+	if (!decode_srp_verifier_and_salt(entry, &salt, &verifier)) {
+		// invalid format, return nil
+		return 0;
```

Comment:
Don't use a "true, false or nil" return type like this in Lua; it's error prone and confusing and especially risky in an authentication interface.

It's better to use two return values, the first one being a boolean and the second one being an error id or a message string to describe what went wrong when the first return value was false.